src/magnet_uri.cpp

```diff
@@ -230,6 +230,33 @@ namespace libtorrent
 			return;
 		}
 
+		std::string::size_type peer_pos = std::string::npos;
+		std::string peer = url_has_argument(uri, "x.pe", &peer_pos);
+		while (!peer.empty())
+		{
+			std::string::size_type divider = peer.find_last_of(':');
```

Comment:
I believe you could use parse_endpoint() https://github.com/arvidn/libtorrent/blob/master/src/socket_io.cpp#L86